cranelift-frontend/src/frontend.rs

```diff
@@ -254,14 +250,6 @@ impl<'a> FunctionBuilder<'a> {
     /// successor), the block will be declared filled and it will not be possible to append
     /// instructions to it.
     pub fn switch_to_block(&mut self, ebb: Ebb) {
-        // First we check that the previous block has been filled.
-        debug_assert!(
-            self.position.is_default()
-                || self.is_unreachable()
-                || self.is_pristine()
-                || self.is_filled(),
-            "you have to fill your block before switching"
-        );
```

Comment:
Without this, there is no check that all switched to ebbs are filled. Also would it be possible to instead of switch to an unfilled block, create a new ebb, add a jump to it and then switch to the target ebb and finally continue with the new ebb if necessary?

```
ebb0:
; some code
; need to switch to ebb1
; create ebb2
jump ebb2

ebb1:
; some other code

ebb2:
; switched to ebb2 instead of ebb0
```

Reply:
> Without this, there is no check that all switched to ebbs are filled. 

Shouldn't that sort of check live in the verifier? I can write a pass as part of this PR if it doesn't exist already.

> instead of switch to an unfilled block, create a new ebb

Would that dead code be eliminated by the optimizer?

Reply:
> Shouldn't that sort of check live in the verifier? I can write a pass as part of this PR if it doesn't exist already.

After adding one instruction, it does. When there is no instruction added yet, it doesnt however.

> Would that dead code be eliminated by the optimizer?

Not in all cases.

Reply:
I added a verifier pass for empty EBBs.